src/data/reports/mal-data-subscription/MalDataSubscriptionDefaultRepository.ts

```diff
@@ -79,98 +28,361 @@ export class MalDataSubscriptionDefaultRepository implements MalDataSubscription
+                );
+
+                const section: NamedRef | undefined = _.chain(dataElement.dataSetElements)
+                    .flatMap("dataSet.sections")
+                    .find(section => _.some(section.dataElements, { id: dataElement.id }))
+                    .value();
+
+                return {
+                    dataElementId: dataElement.id,
+                    dataElementName: dataElement.name,
+                    subscription: !!subscriptionValue?.subscribed,
+                    lastDateOfSubscription: subscriptionValue?.lastDateOfSubscription ?? "",
+                    section,
+                    dataElementGroups: dataElement.dataElementGroups,
+                };
             })
-        );
-
-        return { pager, objects: items };
-    }
-
-    async getColumns(namespace: string): Promise<string[]> {
-        const columns = await this.storageClient.getObject<string[]>(namespace);
+            .filter(row => {
+                const isSubscribed = !!(!subscriptionStatus
+                    ? row
+                    : (subscriptionStatus === "Subscribed") === row.subscription);
+                const isInSection = !!(_.isEmpty(sectionIds) ? row : _.includes(sectionIds, row.section?.id));
+                const isInDataElementGroup = !!(_.isEmpty(dataElementGroupIds)
+                    ? row
+                    : _.intersection(
+                          dataElementGroupIds,
+                          row.dataElementGroups.map(dataElementGroup => dataElementGroup.id)
+                      ).length > 0);
+
+                return isSubscribed && isInSection && isInDataElementGroup;
+            });
 
-        return columns ?? [];
+        const sections = _(rows)
+            .map(row => row.section)
+            .compact()
+            .uniqBy("id")
+            .value();
+
+        const dataElementGroups = _(rows)
+            .flatMap(row => row.dataElementGroups)
+            .uniqBy("id")
+            .value();
+
+        const objects = _(rows)
+            .orderBy([row => row[sorting.field]], [sorting.direction])
+            .drop((paging.page - 1) * paging.pageSize)
+            .take(paging.pageSize)
+            .value();
+
+        const pager: Pager = {
+            page: paging.page,
+            pageSize: paging.pageSize,
+            pageCount: Math.ceil(rows.length / paging.pageSize),
+            total: rows.length,
+        };
+
+        return { pager, objects, sections, dataElementGroups, totalRows: rows };
     }
 
-    async saveColumns(namespace: string, columns: string[]): Promise<void> {
-        return this.storageClient.saveObject<string[]>(namespace, columns);
-    }
+    async getChildrenDataElements(
+        options: MalDataSubscriptionOptions
+    ): Promise<MalSubscriptionPaginatedObjects<DashboardSubscriptionItem>> {
+        const { dashboardSorting, subscriptionStatus, elementType, paging } = options;
+        if (!dashboardSorting)
```

Comment:
Also here to improve performance and avoid unnecessary api calls here you can check also `elementType === "dataElements"`